embark-consult.el

```diff
@@ -103,44 +111,89 @@
+          (setq last-buf this-buf))
+        (insert (concat lineno contents nl))))))
+
+(defun embark-consult--location-to-grep-results-writer (lines)
+  "Format the results from `consult-location' range of commands for the `grep-mode' as alternative to occur.
+You can edit results with `wgrep' in grep buffer including line deletion and insertion of new lines.
+The elements of LINES are assumed to be values of category `consult-location'."
+  (let (last-buf
+        filename)
+    (insert (propertize "Caution:" 'wgrep-header t 'font-lock-face '(:inherit 'compilation-warning :underline t))
+            (propertize " only buffers backed by a file are exported to grep-mode.\n" 'wgrep-header t 'font-lock-face '(:inherit 'compilation-warning)))
+    (insert (propertize "Hint:" 'font-lock-face '(:inherit 'compilation-info :underline t))
+            (propertize " use wgrep for editing of the results in place and saving them to respective files\n" 'font-lock-face '(:inherit 'compilation-info)))
+    (dolist (line lines)
+      (pcase-let*
+          ((`(,loc . ,num) (consult--get-location line))
+           (lineno (format "%d" num))
+           (contents (embark-consult--strip line))
+           (this-buf (marker-buffer loc)))
+        (when (buffer-file-name this-buf)
           (unless (eq this-buf last-buf)
-            (insert (propertize
-                     (format "lines from buffer: %s\n" this-buf)
-                     'face list-matching-lines-buffer-name-face))
-            (setq last-buf this-buf))
-          (insert (concat lineno contents nl))))
+            (setq last-buf this-buf)
+            (setq filename (file-name-nondirectory (buffer-file-name this-buf)))
+            (insert "\n" (propertize (concat "file: " filename) 'wgrep-ignore t 'font-lock-face '(:inherit compilation-info :underline t)) "\n"))
+          (insert (propertize
+                   (concat filename ":")
+                   'invisible t)
+                  lineno
+                  ":"
+                  contents "\n" ))))))
+
+(defun embark-consult--setup-occur-mode (lines resutls-writer)
+  "Create a buffer and configure `occur-mode' and call occur results writer"
+  (let ((buf (generate-new-buffer "*Embark Export Occur*")))
+    (with-current-buffer buf
+      (funcall resutls-writer lines)
       (goto-char (point-min))
       (occur-mode))
     (pop-to-buffer buf)))
 
+(defun embark-consult-export-location (lines)
+  "Create an occur or grep mode buffer listing LINES.
+The elements of LINES are assumed to be values of category `consult-location'."
+  (if embark-consult-export-location-candidates-to-grep-mode
+      (funcall #'embark-consult--setup-grep-mode
+               "Exported Consult results\n\n"
+               lines
+               #'embark-consult--location-to-grep-results-writer)
+    (funcall #'embark-consult--setup-occur-mode
+             lines
+             #'embark-consult--location-to-occur-results-writer)))
```

Comment:
Instead of the customization variable, we should offer two different exporter functions:

embark-consult-export-location-grep
embark-consult-export-location-occur